### PR TITLE
Update component status page

### PR DESCRIPTION
**Pull request by araiford**
This PR touches a lot of files because I had to add a few front-matter attributes to several markdown files to support the helix-ui code indicators, and also to get the design status noted on some articles that didn't have it.  

There are conflicts, but I'll resolve them once this gets the thumbs up.

### CODE REVIEW

File: status.html [hunk 1]
```diff
@@ -1,124 +1,84 @@
...
-                    <i class="icon-check-mark"></i>
-                  {%- elsif page.status == "deprecated" -%}
-                    <i class="icon-error"></i>
-                  {%- endif -%}
-                {%- else -%}
-                  <i class="icon-disabled"></i>
-                {%- endif -%}
+          {% endif %}
+      {% endfor %}
+      {% for page in site.component_design %}
```
Comment (nimbinatus) [inline status.html]: Minor nit: It's a bit odd to have the alphabetical listing restart here. Can we add a table spanner above each block that says something like "General Patterns" for the docs and "Component Patterns" for the components?

File: _component_design/alert-bar.md [hunk 2]
```diff
@@ -6,9 +6,11 @@ category: Components
 usage: >
   A notification bar extends fully across the top of an content container within a Helix interface in order to provide the user with feedback relating to the status of the item. It can be scoped across the entirity of the content area (page level scoping) or scoped to extend across a singular section of the content area (element level scoping).
 preview-image: preview-images/alert-bar.svg
-status: deprecated, in-progress
+status: in-progress
 resource: true
 last-modified: 2017-08-17
+helix-ui-css: true
```
Comment (nimbinatus) [inline _component_design/alert-bar.md]: Long-term thought: We might be able to just query their site's source to find relevant components (and populate the status page with components that don't have a design page). Something to file away. Not sure how we'd do it yet, but you never know :)

File: status.html (hunk 1 above)
Comment (nimbinatus) [inline status.html]: Also, the tables pages kinda throw off the listing. Maybe we should change their filenames from, for example, `table-basic.md` to `basic-table.md` to get this listing right?

File: _component_design/alert-bar.md (hunk 2 above)
Comment (araiford) [reply _component_design/alert-bar.md]: I agree.  I could see us harmonizing the names of our patterns so there's more of a 1:1 relationship -- at least when the code exists.  I also wonder if the helix-ui site could render an RSS feed we could consume, for things like this?

Comment (nimbinatus) [reply _component_design/alert-bar.md]: Hmmm... that could work. There's a lot of plugins to do stuff like that for Jekyll, at least; I imagine whatever platform they're on currently has a similar system.

File: status.html (hunk 1 above)
Comment (araiford) [reply status.html]: agreed.

File: status.html [hunk 3]
```diff
@@ -1,124 +1,84 @@
...
+        <th class="align-center">Design Doc</th>
+        <th class="align-center">CSS</th>
+        <th class="align-center">JS</th>
       </tr>
     </thead>
     <tbody>
-      {%- for page in site.pages -%}
-          {%- if page.category -%}
-              <tr>
+      {% for page in site.docs %}
```
Comment (nimbinatus) [inline status.html]: Try this for lines 22-66 instead:
```nunjucks
{%- for page in site.documents -%}
  {%- if page.resource == true and page.code-element == true -%}
    <tr>
      <td>
        <a href="{{site.baseurl}}{{page.url}}.html">{{page.title}}</a>
      </td>
      <td class="align-center">
        {%- if page.status -%}
          <span class="status-label small {{page.status | replace: ' ','-'}}">{{page.status}}</span>
        {%- endif -%}
      </td>
      <td class="align-center status-ok">
        {%- if page.helix-ui-css -%}
          <hx-icon type="checkmark-circle"></hx-icon>
        {%- endif -%}
      </td>
      <td class="align-center status-ok">
        {%- if page.helix-ui-javascript -%}
          <hx-icon type="checkmark-circle"></hx-icon>
        {%- endif -%}
      </td>
    </tr>
  {%- endif -%}
{%- endfor -%}
```

File: status.html [hunk 4]
```diff
@@ -1,124 +1,69 @@
...
-                    <span class="component deprecated">Deprectated</span>
-                  {%- endif -%}
-                {%- else -%}
-                  <span class="component prototype">In Progress</span>
-                {%- endif -%}
-                </td>
-              </tr>
-          {%- endif -%}
-      {%- endfor -%}
+      {% assign component_list = "" | split: ','%}
```
Comment (nimbinatus) [inline status.html]: minor nit: If you use a hyphen inside the percent signs (`{%- ... -%}`), the resulting pages don't have a ton of whitespace in the source code.

File: _data/design-status.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+ class: stable
+ description: 'The Helix team has a high degree of confidence in this design and considers it low risk and ready for production implementation.'
+in-progress:
+ name: in progress
+ class: 'in-progress'
+ description: 'The Helix team has moderate confidence in this design and considers it medium risk for implementation.  The pattern will likely undergo some changes in the near future.'
+deprecated:
+ name: deprecated
+ class: 'deprecated'
+ description: 'This design is going to be phased out. Helix considers this high risk and the item should no longer be implemented.'
```
Comment (nimbinatus) [inline _data/design-status.yml]: s/`Helix considers this high risk and the item should no longer be implemented.`/`The Helix team considers this design high risk, and this design should no longer be implemented.` or `The Helix team considers this design high risk; do not implement this design.`

File: _data/design-status.yml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+stable:
+ name: stable
+ class: stable
+ description: 'The Helix team has a high degree of confidence in this design and considers it low risk and ready for production implementation.'
```
Comment (nimbinatus) [inline _data/design-status.yml]: suggestion: `The Helix team has a high degree of confidence in this design and considers it low risk. It is ready for production implementation.`

File: status.html (hunk 4 above)
Comment (araiford) [reply status.html]: oh man!  I did not know that!  Thanks for the Pro-Tip!

**Comment by nimbinatus** [comment]
@araiford something weird happened with your rebase, it looks like... Still some conflicts, and some duplicate commits?

**Pull request merged**